satpy/resample/__init__.py

```diff
@@ -0,0 +1,66 @@
+# Copyright (c) 2015-2025 Satpy developers
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+"""Writers subpackage."""
+from __future__ import annotations
+
+import warnings
+from typing import Any
+
+
+def __getattr__(name: str) -> Any:
+    if name in ("KDTreeResampler", "BilinearResampler"):
+        from . import kdtree
+
+        new_submod = "kdtree"
+        obj = getattr(kdtree, name)
+    elif name == "NativeResampler":
+        from .native import NativeResampler
+        new_submod = "native"
+        obj = NativeResampler
+    elif name in (
+            "BucketResamplerBase",
+            "BucketAvg",
+            "BucketSum",
+            "BucketCount",
+            "BucketFraction"
+    ):
+        from . import bucket
+        new_submod = "bucket"
+        obj = getattr(bucket, name)
+    elif name in (
+            "hash_dict",
```

Comment:
Is `hash_dict` used anywhere else? Should it be "private-ized"? It would need handling in this if statement, but better to make things private now rather than later.

Reply:
Didn't find any calls to it outside the new "base" module, so made it private and removed from the `__getattr__` handling in https://github.com/pytroll/satpy/pull/3124/commits/44ee39b9c122da80218cebfd5ed078ecf4f495e2